aip/general/0193.md

```diff
@@ -75,132 +84,190 @@ following:
 }
 ```
 
-### Details
-
-Google defines a set of [standard detail payloads][details] for error details,
-which cover most common needs for API errors. Services **should** use these
-standard details payloads when feasible.
+#### google.rpc.Status
 
-Structured details with machine-readable identifiers **must** be used if users
-will need to write code against a specific aspect of the error. Error message
-strings **may** change over time; however, if an error message does not have a
-machine-readable identifier _in addition to_ the `code` field, changing the
-error message **must** be considered a backwards-incompatible change.
+Services must return a [`google.rpc.Status`][Status] message when an API
+error occurs. For each error the service **must** populate the `message`
+string in [`google.rpc.Status`][Status] as follows:
 
-#### ErrorInfo
-
-The [`ErrorInfo`][ErrorInfo] message is the required way to send a
-machine-readable identifier. All error responses **must** include an
-`ErrorInfo` payload in the `details` field. Variable information
-**should** be included in the `metadata` field on `ErrorInfo` and
-**must** be included if it appears within an error message.
++ present a developer-facing, human-readable "debug message"
++ present the message in the service's native language to explain both
+  the error and offer an actionable resolution to it ([citation][Error
+  model])
 
-#### Uniqueness
+**Note:** Sometimes a service will elect to always present
+`Status.message` in English rather than the application's native
+language so that messages are easily searchable in common knowledge
+bases, such as StackOverflow™.
+
+Important points:
+
++ This `message` string is considered a problem description. It is
+  intended for developers to understand the problem and is more
+  detailed than [`ErrorInfo.reason`][ErrorInfo-reason], discussed in the
+  next section.
++ Use simple descriptive language that is easy to understand (without
+  technical jargon) to clearly state the problem that results in an
+  error.
++ For pre-existing (brownfield) APIs, the information in
+  `google.rpc.Status` already exists and can’t be changed. In this
+  case, do not edit anything in this `google.rpc.Status`, because some
+  users might have automation running against the pre-existing content.
+  For example, any changes in text for the `message` string could cause
+  breakages. For more information, see [Changing Error
+  Messages](#changing-error-messages).
+
+##### [google.rpc.Status payload][Status]
+
+Details of this object are summarized in the following fields, field
+descriptions, and examples:
+
++ `code int32`
+  + The status code, which **must** be an enum value of
+    [`google.rpc.Code`][Code].
+  + *Example*: 429
```

Comment:
this example is wrong, google.rpc.Code is different from HTTP status code. In this case, it should be `RESOURCE_EXHAUSTED`: https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto#L108